httpx/error.go

```diff
@@ -28,27 +24,18 @@ type Error struct {
 	StatusCode int
 }
 
-// errorResponse is an error response.
-type errorResponse struct {
-	Message string `json:"message"`
-}
-
-// Write observes the values of the error and writes it to the http.Writer.
-func (e *Error) Write(ctx context.Context, w http.ResponseWriter) {
-	obs := observe.FromContext(ctx)
-
-	defer obs.Error(ctx, e.Message, e.Err)
-
-	w.Header().Set("Content-Type", jsonContentType)
-
-	code := e.StatusCode
-	if code <= 0 {
-		code = http.StatusInternalServerError
+// GetStatusCode returns the status code for the error.
+func (e Error) GetStatusCode() int {
```

Comment:
I think you can dry this up by leveraging `Response`:
```
func NewErrorResponse(status int, err error, msg string) Writeable {
	if status == 0 {
		status = 500
	}

	if err != nil && msg != "" {
		msg = fmt.Sprintf("%s: %s", msg, err)
	} else if err != nil {
		msg = err.Error()
	} else if msg == "" {
		msg = http.StatusText(status)
	}
	
	return Response{
		Payload:    MessagePayload{Message: msg},
		StatusCode: status,
	}
}

type MessagePayload struct {
	Message string `json:"message"`
}
```
Might be good to have a way to `head` a status code with no body too. 

Also how message gets composed seems a little convoluted to have in here IMO. Maybe we just pass in message and can have helpers to help the caller compose a message or just set the default in here

Reply:
For `HEAD` (as an example) you don't need anything more than

```go
w.WriteHeader(http.StatusOK)
```